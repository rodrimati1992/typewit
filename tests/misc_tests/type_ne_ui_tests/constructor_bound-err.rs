use typewit::TypeNe;

fn unconstrained_with_any_call<T, U>() -> Option<TypeNe<T, U>> {
    TypeNe::<T, U>::with_any()
}



fn main() {}