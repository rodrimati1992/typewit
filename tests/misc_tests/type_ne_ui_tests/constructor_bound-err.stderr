warning: use of deprecated associated function `typewit::type_ne_::<impl typewit::TypeNe<L, R>>::with_any`: fallout of `https://github.com/rust-lang/rust/issues/97156`,`TypeId::of::<L>() != TypeId::of::<R>()` does not imply `L != R`
 --> tests/misc_tests/type_ne_ui_tests/constructor_bound-err.rs:4:21
  |
4 |     TypeNe::<T, U>::with_any()
  |                     ^^^^^^^^
  |
  = note: `#[warn(deprecated)]` on by default

error[E0310]: the parameter type `T` may not live long enough
 --> tests/misc_tests/type_ne_ui_tests/constructor_bound-err.rs:4:5
  |
4 |     TypeNe::<T, U>::with_any()
  |     ^^^^^^^^^^^^^^^^^^^^^^^^
  |     |
  |     the parameter type `T` must be valid for the static lifetime...
  |     ...so that the type `T` will meet its required lifetime bounds
  |
help: consider adding an explicit lifetime bound
  |
3 | fn unconstrained_with_any_call<T: 'static, U>() -> Option<TypeNe<T, U>> {
  |                                 +++++++++

error[E0310]: the parameter type `U` may not live long enough
 --> tests/misc_tests/type_ne_ui_tests/constructor_bound-err.rs:4:5
  |
4 |     TypeNe::<T, U>::with_any()
  |     ^^^^^^^^^^^^^^^^^^^^^^^^
  |     |
  |     the parameter type `U` must be valid for the static lifetime...
  |     ...so that the type `U` will meet its required lifetime bounds
  |
help: consider adding an explicit lifetime bound
  |
3 | fn unconstrained_with_any_call<T, U: 'static>() -> Option<TypeNe<T, U>> {
  |                                    +++++++++
